toolchain/sem_ir/typed_insts.h

```diff
@@ -666,9 +669,9 @@ struct ImportRefLoaded {
 // `src_id`, by performing a final copy from source to destination, for types
 // whose initialization is not in-place.
 struct InitializeFrom {
-  // TODO: Make Parse::NodeId more specific.
-  static constexpr auto Kind = InstKind::InitializeFrom.Define<Parse::NodeId>(
-      {.ir_name = "initialize_from"});
+  static constexpr auto Kind =
+      InstKind::InitializeFrom.Define<Parse::InvalidNodeId>(
```

Comment:
As I recall, the point of the typed node ID parameter here was to guarantee that the node ID for this instruction will always be of the given type. If we specify it as being `InvalidNodeId`, I think that means the node ID for this instruction is always invalid, that is, it has no associated parse node. That doesn't seem to be the case for `InitializeFrom` -- instead, it looks like it can be associated with any parse node, so `NodeId` seems right to me.

Reply:
I could do that, but we don't built it using a node ID. `Convert` is constructing it with a `LocId`, and looking, I don't ever see a parse node directly used for a `Convert` call. So is the preference for, when it's never constructed with a NodeId, to still support constructing with an arbitrary NodeId just-in-case? Maybe that's something I can just put in a comment, to reflect that it's not validated?